apt-hook/20apt-esm-hook.conf

```diff
@@ -1,8 +1,10 @@
 APT::Update::Post-Invoke-Stats {
+        "[ ! -f /usr/lib/ubuntu-advantage/ua_update_messaging.py ] || python3 /usr/lib/ubuntu-advantage/ua_update_messaging.py";
```

Comment:
Yes, now that these pieces are fitting together, it looks like we might need to rethink parts of this and see what the best solution is. I like the speed of C++ esm-apt-hook here, we probably shouldn't be calling it twice though, once from python, once with the hook. This is the rub with baking our python ua_message_update into apt hooks instead of having a stand-alone systemd timer. We should probably ping the broader team on this to see if there are some gotchas. I know folks are not happy with apt hooks doing much more than simple file reads hence I didn't try to refactor these two modules yet because I think we may have to consider a separate systemd unit or timer.

Reply:
I put up a sample set of commits for us to talk about Monday.
apt-hook pkg-status-json d61f30a513ed3de978e7df5a2b525d6d2b4db1b2 and ua status unattached messaging about 11ad82aaec4ed0851037639a8ba54c0643061037